src/util.js

```diff
@@ -406,19 +406,35 @@
    */
   connect.PopupManager = function () { };
 
-  connect.PopupManager.prototype.open = function(url, name) {
+  connect.PopupManager.prototype.DEFAULT_WINDOW_HEIGHT = 578;
+  connect.PopupManager.prototype.DEFAULT_WINDOW_WIDTH = 433;
+
+  connect.PopupManager.prototype.open = function (url, name, options) {
     var then = this._getLastOpenedTimestamp(name);
     var now = new Date().getTime();
-    var win = null;      
+    var win = null;
     if (now - then > ONE_DAY_MILLIS) {
-       win = window.open('', name);
-       if (win.location !== url) {
+      if (options && options.forceWindow === true) {
+        // default values below are chosen to provide a minimum height without scrolling
+        // and a unform margin based on the css of the ccp login page
+        var height = options.height ? options.height : this.DEFAULT_WINDOW_HEIGHT;
+        var width = options.width ? options.width : this.DEFAULT_WINDOW_WIDTH;
+        var y = isNaN(options.top) ? window.top.outerHeight / 2 + window.top.screenY - (height / 2) : options.top;
```

Comment:
After looking at a couple ways to do it, I went with

```js
var top = isNaN(options.top) ? 0 : options.top
```

to avoid having to build the window features string on the fly, and still have some fallback behavior if the caller gives bad/no value. I think letting users center it themselves is reasonable.

Reply:
Sorry I made a comment on this before reading your explanation here.
I think it's very good of you to check but since you have clear documentation already here, we can just leave it to the clients to work out bad inputs :)